src/uu/date/src/date.rs

```diff
@@ -212,11 +230,14 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         return set_system_datetime(date);
     } else {
         // Get the current time, either in the local time zone or UTC.
-        let now: DateTime<FixedOffset> = if settings.utc {
-            let now = Utc::now();
-            now.with_timezone(&now.offset().fix())
+        let now = get_current_time(settings.utc);
+        // Convert to FixedOffset for consistent timezone handling
+        let now_fixed = if should_use_utc(settings.utc) {
```

Comment:
Why aren't you moving the `now_fixed` logic into `get_current_time`? Seems like you output something with the wrong timezone in `get_current_time` that you now need to adjust again?